portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -15,13 +21,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Get the values entered in the form.
     String firstName = request.getParameter("first-name");
     String lastName = request.getParameter("last-name");
+    String fullName = firstName + " " + lastName;
     String email = request.getParameter("email");
     String message = request.getParameter("message-content");
 
-    // Print the value so you can see it in the server logs.
-    System.out.println(firstName + " " + lastName + " wants to contact you:");
-    System.out.println("Their email is: " + email);
-    System.out.println("Their message was: " + message);
+    //Store values in Datastore
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Message");
+    FullEntity messageEntity =
+        Entity.newBuilder(keyFactory.newKey())
```

Comment:
Probably key should be provided by you in case you want to retrieve messages in the future by the specific key.